lib/Onyx.js

```diff
@@ -1050,6 +1050,13 @@ function clear(keysToPreserve = []) {
             const keyValuesToReset = [];
             const defaultKeys = _.keys(defaultKeyStates);
 
+            // Get all the values for the keys that need to be preserved. These key/value pairs will be set
+            // in Onyx after the database is cleared().
+            const keyValuesToPreserve = _.reduce(keysToPreserve, (finalArray, keyToPreserve) => {
+                finalArray.push([keyToPreserve, cache.getValue(keyToPreserve)]);
+                return finalArray;
+            }, []);
+
```

Comment:
Instead of using `_.reduce` it's simpler to just use `_.map`

```js
const keyValuesToPreserve = _.map(keysToPreserve, key => [key, cache.getValue(key)]);
```

We remap every key we want to preserve to a key-value pair, in the end we still end up with the same list of key-value pairs

Reply:
Agreed, makes things much easier to read.